### PR TITLE
Storybook: Added Blockies Package

### PR DESCRIPTION
**Pull request by Douglashdaniel**
Resolves <https://github.com/brave/brave-browser/issues/15099>

Added Blockies Package and added to Connect/Connected Panels.

New Package: https://github.com/jamesmudgett/blockies.git

Ran `npm audit` and it passed!

Will need to run `npm install`

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. `npm run storybook`
2. Confirm that storybook shows the new component.
3. Check functionality of the new component.

https://user-images.githubusercontent.com/40611140/114894869-3d2a3e80-9dcc-11eb-8940-e23d17427019.mov

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -277,6 +277,7 @@
     "emptykit.css": "1.0.1",
     "enzyme": "3.11.0",
     "enzyme-adapter-react-16": "1.15.6",
+    "ethereum-blockies": "git+https://github.com/jamesmudgett/blockies.git",
```
Comment (diracdeltas) [inline package.json:280]: i would suggest pinning to a specific commit or tag here. also some questions:

1.  can we move this repo from james' personal account to the brave github org?
2. how do we plan to keep this repo up-to-date with important fixes in upstream such as security fixes?

Comment (Douglashdaniel) [reply package.json]: Thank you for the suggestions! The repo has been moved here <https://github.com/brave/blockies> and is also now pinned to the latest commit. As far as keeping the repo up-to-date, this repo does not rely on any dependancies at all. It's basically just a JS function that returns value, so I don't believe that will be an issue. :)

File: components/brave_wallet_ui/components/extension/connected-panel/index.tsx [hunk 2]
```diff
@@ -45,8 +46,10 @@ export default class ConnectedPanel extends React.PureComponent<Props> {
   render () {
     const FiatBalance = this.props.selectedAccount.balance * 2000
     const { selectedAccount, isConnected, connectAction, navAction } = this.props
+    const bg = background({ seed: selectedAccount.address })
+    const orb = create({ seed: selectedAccount.address, size: 8, scale: 16 }).toDataURL()
```
Comment (petemill) [inline components/brave_wallet_ui/components/extension/connected-panel/index.tsx:50]: This seems computationally expensive to run on every re-render. I suggest wrapping in a React Hook, like `useMemo` which would only get called again if `selectedAddress.address` changes.

File: components/brave_wallet_ui/components/extension/select-address/index.tsx [hunk 3]
```diff
@@ -25,10 +26,11 @@ export interface Props {
 export default class SelectAddress extends React.PureComponent<Props> {
   render () {
     const { account, isSelected, action } = this.props
+    const orb = create({ seed: account.address, size: 8, scale: 16 }).toDataURL()
```
Comment (petemill) [inline components/brave_wallet_ui/components/extension/select-address/index.tsx:29]: same issue re: performance

File: components/definitions/ethereum-blockies.d.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+declare module 'ethereum-blockies'
```
Comment (petemill) [inline components/definitions/ethereum-blockies.d.ts:1]: What's the point of this if it's empty? Can we add the basic API signature anyway?

**Review by petemill — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/components/extension/connected-panel/index.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/connected-panel/index.tsx]: Updated this to be a functional component and changed to useMemo Hook!

File: components/brave_wallet_ui/components/extension/select-address/index.tsx (hunk 3 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/extension/select-address/index.tsx]: Updated this to be a functional component and changed to useMemo Hook!

File: components/definitions/ethereum-blockies.d.ts (hunk 4 above)
Comment (Douglashdaniel) [reply components/definitions/ethereum-blockies.d.ts:1]: The reason I added this was to get rid of this type declaration error.

Screen Shot 2021-04-16 at 6 50 04 PM

**Review by petemill — APPROVED** [review]

**Pull request merged**